filebeat/input/filestream/internal/input-logfile/store_test.go

```diff
@@ -31,13 +31,28 @@ import (
 	"github.com/elastic/beats/v7/libbeat/logp"
 	"github.com/elastic/beats/v7/libbeat/statestore"
 	"github.com/elastic/beats/v7/libbeat/statestore/storetest"
+	"github.com/elastic/go-concert/unison"
 )
 
 type testStateStore struct {
 	Store    *statestore.Store
 	GCPeriod time.Duration
 }
 
+func TestResource_CopyInto(t *testing.T) {
+	src := resource{lock: unison.MakeMutex()}
+	dst := resource{lock: unison.MakeMutex()}
+	src.lock.Lock()
+	dst.lock.Lock()
```

Comment:
Do we want to do anything about the case where `src` is unlocked but `dst` is unlocked?

I don't think that is possible in the code today, but maybe it will be someday.

Reply:
The point of the test is not to check the locks but to check the field is not overwritten. It's just a good way to test it. Also, it's a regression test and this how it panicked before.